mod-source-record-manager-server/src/main/java/org/folio/services/flowcontrol/RawRecordsFlowControlService.java

```diff
@@ -0,0 +1,33 @@
+package org.folio.services.flowcontrol;
+
+/**
+ * Service to implement flow control logic to be able to import OCLC files in between imports of huge files.
+ * Its necessary to not wait for importing 1 record file until other big files in progress.
+ * Its can be suitable not only for OCLC, but also for other files that skipping DI_RAW_RECORDS_CHUNK_READ stage
+ * and directly pushing messages to DI_RAW_RECORDS_CHUNK_PARSED topic.
+ */
+public interface RawRecordsFlowControlService {
+
+  /**
+   * Tracks each DI_RAW_RECORDS_CHUNK_READ event, this method can also pause processing
+   * of DI_RAW_RECORDS_CHUNK_READ topic when flow control conditions met.
+   *
+   * @param initialRecordsCount records count in the batch
+   */
+  void trackChunkReceivedEvent(Integer initialRecordsCount);
```

Comment:
would this method be tracking one event or many events? This method could be plural:`trackChunkReceivedEvents`

Reply:
if this is acceptable, other methods in this interface should be plural as well.

Reply:
Reading the code a little bit more, I think if this method is named this way, it would accept the event as a parameter. In the implementation, the initial record count can be gotten.

or the name of these methods should be like `incrementChunkReceivedCount(int recordCount)`

Reply:
we receiving messages from DI_RAW_RECORDS_CHUNK_READ topic in chunks, it is one event that contain chunk with multiple records. For DI_COMPLETE/DI_ERROR we always operate with individual records(not chunks).
Based on this we have words 'chunk' and 'record' in method names.
I have not added something like 'count' in method name to not associate it with simple counter. 
I have not added event as parameter to simplify things, if we not use this object in method - we dont need to construct full event and pass as param.
@okolawole-ebsco  what do you think, can we remain method names as it or you have additional suggestions?

Reply:
I dont have any other suggestions. Your explanation helped and the changes you made seems more correct to me than the previous version.